Beams/Logic/vtkSlicerBeamsModuleLogic.cxx

```diff
@@ -299,3 +296,195 @@ void vtkSlicerBeamsModuleLogic::ProcessMRMLNodesEvents(vtkObject* caller, unsign
+  // so that UpdateTransformsFromBeamGeometry is called. It may be needed to change the signature of the
+  // update function. It may be also needed to store a reference to the beam node (see defined nodes in SlicerRT)
+
+  if (!beamNode)
+  {
+    vtkErrorMacro("UpdateBeamTransform: Invalid beam node");
+    return;
+  }
+
+  // Make sure transform node exists
+  beamNode->CreateDefaultTransformNode();
+
+  // Update transform for beam
+  vtkMRMLLinearTransformNode* beamTransformNode = vtkMRMLLinearTransformNode::SafeDownCast(
+    beamNode->GetParentTransformNode());
+  if (!beamTransformNode)
+  {
+    vtkErrorMacro("UpdateBeamTransform: Failed to access transform node of beam " << beamNode->GetName());
+    return;
+  }
+
+  this->UpdateBeamTransform(beamNode, beamTransformNode);
+}
+
+//-----------------------------------------------------------------------------
+void vtkSlicerBeamsModuleLogic::UpdateBeamTransform(vtkMRMLRTBeamNode* beamNode, vtkMRMLLinearTransformNode* beamTransformNode, double* isocenter/*=nullptr*/)
+{
+  if (!beamNode)
+  {
+    vtkErrorMacro("UpdateBeamTransform: Invalid beam node");
+    return;
+  }
+  if (!beamTransformNode)
+  {
+    vtkErrorMacro("UpdateBeamTransform: Invalid beam transform node");
+    return;
+  }
+
+  // Update transforms in IEC logic from beam node parameters
+  this->UpdateIECTransformsFromBeam(beamNode, isocenter);
+
+  // Dynamic transform from Collimator to World
+  vtkMRMLLinearTransformNode* collimatorToGantryTransformNode = this->GetTransformNodeBetween(vtkSlicerIECTransformLogic::Collimator, vtkSlicerIECTransformLogic::Gantry);
+  vtkNew<vtkGeneralTransform> beamGeneralTransform;
+  collimatorToGantryTransformNode->GetTransformToWorld(beamGeneralTransform);
+
+  // Convert general transform to linear
+  // This call also makes hard copy of the transform so that it doesn't change when other beam transforms change
+  vtkNew<vtkTransform> beamLinearTransform;
+  if (!vtkMRMLTransformNode::IsGeneralTransformLinear(beamGeneralTransform, beamLinearTransform))
+  {
+    vtkErrorMacro("UpdateBeamTransform: Unable to set transform with non-linear components to beam " << beamNode->GetName());
+    return;
+  }
+
+  // Set transform to beam node
+  beamTransformNode->SetAndObserveTransformToParent(beamLinearTransform);
+}
+//-----------------------------------------------------------------------------
+vtkMRMLLinearTransformNode* vtkSlicerBeamsModuleLogic::GetTransformNodeBetween(
+  vtkSlicerIECTransformLogic::CoordinateSystemIdentifier fromFrame, vtkSlicerIECTransformLogic::CoordinateSystemIdentifier toFrame)
+{
+  if (!this->GetMRMLScene())
+  {
+    vtkErrorMacro("GetTransformNodeBetween: Invalid MRML scene");
+    return nullptr;
+  }
+
+  return vtkMRMLLinearTransformNode::SafeDownCast(
+    this->GetMRMLScene()->GetFirstNodeByName(this->IecLogic->GetTransformNameBetween(fromFrame, toFrame).c_str()));
+}

```

Comment:
Empty line needed. Please check everywhere. And in general, always make sure that the coding style matches the rest of the code. New code should not be incompliant. Thanks